src/contexts/ModuleContext.tsx

```diff
@@ -1,41 +1,94 @@
-/* eslint-disable no-console */
-/* eslint-disable @typescript-eslint/naming-convention */
-/* eslint-disable @typescript-eslint/no-explicit-any */
 import {
   PropsWithChildren,
   ReactElement,
   createContext,
+  useContext,
   useEffect,
   useState,
 } from 'react';
 
 import { OctoModule, manager } from '@octopost/module-manager';
 
-export interface ModuleInterface {
+import { fetchModules } from '~services/axios/modules';
+
+import { IPluginMetadata } from '../../electron/utils/readPackageJson/readPackageJson.types';
+
+export interface IModuleInterface {
   components: ReactElement[];
 }
 
-export interface ModuleContext {
+export interface IModuleContext {
   modules: OctoModule[];
+  modulesURL: string[];
 }
 
-export const ModuleContext = createContext<ModuleContext | null>(null);
+export const ModuleContext = createContext({} as IModuleContext);
+
+export const useModule = () => useContext(ModuleContext);
```

Comment:
Mover as interfaces para um `types.ts`.

Porque o hook do useContext ta fora do componente? Porque precisamos do `useContext` wrappado no `useModule`?

Reply:
criar um hook pra ser um wrapper do context é uma convenção que eu gosto de seguir quando to mexendo com context, dessa forma eu só preciso importar o `useModule` pra usar as coisas dele, ao contrário do jeito normal que eu teria que importar o `useContext` e o `ModuleContext` pra poder usar esse context. Sobre a posição dele, ele ta fora do componente porque é uma função que precisa ser exportada pra eu poder usar a função importando ela onde eu preciso.